stingray/pulse/pulsar.py

```diff
@@ -263,18 +263,24 @@ def fold_events(times, *frequency_derivatives, **opts):
     profile_err : array of floats
         The uncertainties on the pulse profile
     """
-    mode = _default_value_if_no_key(opts, "mode", "ef")
-    nbin = _default_value_if_no_key(opts, "nbin", 16)
-    weights = _default_value_if_no_key(opts, "weights", 1)
+
+    mode = opts.pop("mode", "ef")
+    nbin = opts.pop("nbin", 16)
+    weights = opts.pop("weights", 1)
     # If no key is passed, *or gti is None*, defaults to the
     # initial and final event
-    gti = _default_value_if_no_key(opts, "gti", None)
+    gti = opts.pop("gti", None)
     if gti is None:
         gti = [[times[0], times[-1]]]
     # Be safe if gtis are a list
     gti = np.asanyarray(gti)
-    ref_time = _default_value_if_no_key(opts, "ref_time", 0)
-    expocorr = _default_value_if_no_key(opts, "expocorr", False)
+    ref_time = opts.pop("ref_time", 0)
+    expocorr = opts.pop("expocorr", False)
+
+    if opts:
+        raise ValueError(
+            f"Unidentified keys: {opts.keys()} \n Please refer to the description of the function for optional parameters."
```

Comment:
`Unidentified keyword to fold_events: {opts.keys()}` etc. would probably be more informative